### DISCUSSION TITLE
Why are DLL's checked into the repo?

### DISCUSSION THREAD

**Discussion by jwdonahue** [stride3d/stride#1856] [Contributors]
I see a lot of executable content checked into the stride repo.  Why?

My first impulse is to dive-in and fix that, but first I need to understand why they were put there in the first place?

**Comment by tebjan** [comment]
These are mainly dependencies that Stride needs to function. Either because the source is not available, or building it would make the build process too complex and hard for contributors to setup the build environment. But some are working to replace some of their functionality with new open source and C# alternatives that didn't exist when they were added many years ago.

**Comment by jwdonahue** [comment]
Any binaries that cannot be replaced, should be pushed to a package feed and removed from the repo.  I generally keep at least two clones of any project repo I am active in, and while I have a lot of disk space, I want to keep it that way.  I would be happy to jump in and help fix or at least mitigates some of these issues.

Is it even possible to call it an open source project if there's no source code available?  I would be really worried about any latent security issues in binaries that can't be patched.  Have you considered decompiling them and pushing the code to ancillary projects/repo's? 

While few in number, every game code base I've encountered so far is full of old bits that can't be properly maintained. Stride seems to be the closest one being nearly pure managed code with reasonably active contributors, but it doesn't seem to work out of the box and I think the messy build system has a lot to do with that.

**Reply by tebjan** [reply]
Your points are absolutely valid, but it's important to note that historically, this project has been managed by a sizable full-time team. As an open-source project, certain issues may not be top priorities due to limited manpower. Resources are often better directed towards issues directly impacting user experience like features, and workflow, where they have a more immediate impact.

**Reply by tebjan** [reply]
But if you have ideas about what to improve, e.g. making packages, you are very welcome to have a look and make a proposal. You can open PRs or issues for discussion without working on them. Or chat with us in the #developer channel on discord.

**Reply by jwdonahue** [reply]
> But if you have ideas about what to improve, e.g. making packages, you are very welcome to have a look and make a proposal. You can open PRs or issues for discussion without working on them.

I do and I will, but I am not a fan of Discord.

> As an open-source project, certain issues may not be top priorities due to limited manpower. Resources are often better directed towards issues directly impacting user experience like features, and workflow, where they have a more immediate impact.

I prefer _customer_ over _user_, but I did arrive here from that perspective. I was looking to see if there was an OSS game engine I could use to write a simple, distributed, two player game, that might save me a few hours work.  I installed Stride and found that it wouldn't let me develop against the latest .NET 6+.  I found some chatter that .NET 5 & 6 were in the works, so I pulled the repo.

I started writing code in 1981 and I've been doing it professionally in organizations big and small for 33 years. It is my experience that deprioritizing code and build hygiene, will take you down a spiraling path to failure. I know there has to be a balance in there, but binaries in the repo gives off a really nasty smell and makes me wonder what other detritus I might find. 

Binary curation and provenance is a problem that's had simple and readily available solutions for more than two decades.

**Reply by tebjan** [reply]
> I installed Stride and found that it wouldn't let me develop against the latest .NET 6+. I found some chatter that .NET 5 & 6 were in the works, so I pulled the repo.

That is odd, did you get the latest package with 1898 in the version? Stride has been on .NET 6 for a long time now. .NET 7 and .NET 8 are also possible from branches, but Stride officially only supports .NET LTS releases.

**Reply by jwdonahue** [reply]
Interesting.  I didn't dig into it at all.  But now I have the repo, I suspect maybe it had something to do with the fact I have VS 2022 and recently added .NET 8 to my list of targets?  I am using the preview and so far it's working quite well with my existing code bases.

I just got this little gem:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	NETSDK1145	The Apphost pack is not installed and NuGet package restore is not supported. Upgrade Visual Studio, remove global.json if it specifies a certain SDK version, and uninstall the newer SDK. For more options visit   https://aka.ms/targeting-apphost-pack-missing  Pack Type:Apphost, Pack directory: C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64, targetframework: net6.0, Pack PackageId: Microsoft.NETCore.App.Host.win-x64, Pack Package Version: 6.0.19	Stride.Importer.FBX	C:\Program Files\dotnet\sdk\8.0.100-preview.6.23330.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets	142	
```
Trying to build against:

commit 9e3b22c4761a8fd02ec7207826c1b46466043235 (HEAD -> master, origin/master, origin/HEAD, wip)
Merge: ce1bdd48e 7eef1ba10
Author: Marian Dziubiak <marian.dziubiak@gmail.com>
Date:   Wed Sep 27 20:16:15 2023 +0100

    Merge pull request #1854 from JeromyWalsh/master

    Replaced 'xcopy' Exec task with the more universal Copy task to suppo…

**Reply by tebjan** [reply]
Do you have .NET 6 SDK installed? You need specifically that one.

**Reply by jwdonahue** [reply]
Ya, I have the latest .NET5, 6, 7 & 8 SDK's installed, but that's really off topic at the moment.  I think I can fix the Apphost problem with a simple PR. I do a lot of multitarget builds, so I insist that code bases I work on, should not require me to uninstall anything or change my defaults.

I started this thread to try and understand what issues were blocking the removal of binaries from the repo. If it's really just neglect, then I should be able to fix that, once I get the current repo to build. I don't usually pull very hard on loose threads in OSS projects. It's hard to know just how deep they go and I was just looking for a distraction with the game idea, not a major project.

**Comment by jwdonahue** [comment]
I guess I need to know where these binaries are coming from?

**Reply by manio143** [reply]
Some of the binaries come from forked projects in the Stride organization. Others may have been placed there a long time ago and are unused or could be replaced with newer alternatives.
